src/raft.rs

```diff
@@ -260,6 +266,9 @@ pub struct RaftCore<T: Storage> {
 
     /// Max size per committed entries in a `Read`.
     pub(crate) max_committed_size_per_ready: u64,
+
+    // Message group cache for follower replication.
+    msg_group: HashMap<u64, Vec<(Message, bool)>>,
```

Comment:
`Vec` is more efficient than `HashMap`.

Reply:
Not just query, but also update and memory footprint.

Reply:
Done. Use  `Vec<(u64, Vec<(Message, bool))>` instead of `HashMap<u64, Vec<(Message, bool)>>`.